File: leetcode/twosum/twosum.py
from typing import List

class twosum:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}

        for i in range(len(nums)): # for i, num in enumerate(nums):
            complement = target - nums[i] # complement = target - num

            if complement in seen:
                return [i, seen[complement]]

            seen[nums[i]] = i # seen[num] = i

File: leetcode/twosum/twosum.java
package leetcode.twosum;

import java.util.HashMap;

class twosum {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> seen = new HashMap<>();

        for(int i = 0; i < nums.length; i++){
            int num = nums[i];
            int complement = target - num;

            if(seen.containsKey(complement)){
                return new int[] {i, seen.get(complement)};
            }

            seen.put(num, i);
        }

        return new int[] {};
    }
}